Refactor classes

ALEXJAZZ008008 opened the pull request:
Refactored and added point cloud output (raw)

ALEXJAZZ008008 committed:
added getters and setters for all member variables in order to add copy and move etc to all classes, ensured all classes meet standards and pass correctly etc

ALEXJAZZ008008 committed:
added ability to output true point clouds ad txt and bin

ALEXJAZZ008008 committed:
moved smoothing process, now can output raw data

NikEfth committed:
fixed cmake files

NikEfth commented on src/KinectFrontend/KinectFrontend.h:
```diff
@@ -43,9 +42,57 @@ class KinectFrontend : public QDialog
+        return m_kinect_interface_ptr;
+    }
+
+    inline int ser_kinect_interface_ptr(KinectInterface * kinect_interface_ptr)
```
typo

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+    {
+        for(unsigned short j = 0; j < KinectBackend::getInstance().m_kinect_object_ptr->get_resolution().at(0); ++j)
+        {
+            KinectBackend::getInstance().m_kinect_object_ptr->get_depth().at(j).at(i) =
```
Could you change this to be a 1D array (or vector)?

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+
+void KinectBackend::depth_callback(freenect_device *m_freenect_device_ptr, void *data, unsigned int timestamp)
+{
+    unsigned short *depth_ptr = (unsigned short *)data;
```
static_cast

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+    return 1;
+}
+
+void KinectBackend::depth_callback(freenect_device *m_freenect_device_ptr, void *data, unsigned int timestamp)
```
check if empty? or the same?

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_got_depth(true);
+
+    KinectBackend::getInstance().append_depth_message("Received depth frame at " + to_string(timestamp) + "\n");
```
Reduce the number of characters.

NikEfth replied:
move at interface

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+
+    KinectBackend::getInstance().m_kinect_object_ptr->set_got_video(true);
+
+    KinectBackend::getInstance().append_video_message("Received video frame at "  + to_string(timestamp) + "\n");
```
same as above

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+    KinectBackend::getInstance().append_video_message("Received video frame at "  + to_string(timestamp) + "\n");
+}
+
+int KinectBackend::calculate_point_cloud()
```
move in the point cloud numerics

NikEfth replied:
If you find the link add it here!

NikEfth commented on src/KinectInterface/KinectBackend/KinectBackend.cpp:
```diff
@@ -0,0 +1,410 @@
+    {
+        for(unsigned short j = 0; j < m_kinect_object_ptr->get_resolution().at(0); ++j)
+        {
+            *vector_iterator = (j - (m_kinect_object_ptr->get_resolution().at(1) / 2.0f)) * (m_kinect_object_ptr->get_depth().at(j).at(i) - 10) * 0.0021f;
```
0.0021f make it a variable. The default value will be set in the constructor. Create a set function

NikEfth replied:
m_kinect_object_ptr->get_resolution().at(1) cast to float

NikEfth committed:
Better Frontend. 

Konnector replaced the Frontend as a manager of connections and acquisitions. It has support for settings which need to be passed in the backend. In addition, now thw connection and acquisition can be decoupled. 

For the time being the motor control is not implemented and commented out.

ALEXJAZZ008008 committed:
added most of the changes to make more proper c++ code, need to change to use vectors etc. maths code moved to pointcloud library

ALEXJAZZ008008 committed:
finished adding constructors and getters, adding output to frontend

ALEXJAZZ008008 committed:
added settings

ALEXJAZZ008008 committed:
changed speed to take input

ALEXJAZZ008008 committed:
refactoring, changed to smart_ptr caused issues with c style

NikEfth committed:
Final (?) comments

ALEXJAZZ008008 committed:
added motor controls

NikEfth committed:
Make CLang happy

ALEXJAZZ008008 committed:
changed angle to take both double but with two methods rather than overload

ALEXJAZZ008008 commented:
Please merge me!

ALEXJAZZ008008 committed:
changed .at() to []
